### PR TITLE
network: Add support for multiq qdisc

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
LGTM.

**Comment by DaanDeMeyer** [comment]
@yuwata I turns out there's both "mq" and "multiq" qdiscs. I'm a bit stuck on the naming. `mq` exposes multiple hardware queues whereas `multiq` exposes multiple software queues on top of a single hardware queue if I understand correctly. So maybe `HardwareMultiQueueing` and `SoftwareMultiQueueing`?

**Comment by yuwata** [comment]
And there is also "mqprio", huh...

**Comment by DaanDeMeyer** [comment]
From the kconfig:

```
config NET_SCH_MULTIQ
	tristate "Hardware Multiqueue-aware Multi Band Queuing (MULTIQ)"
```

So maybe `[MultiBandQueueing]`?

**Comment by yuwata** [comment]
From sch_multiq.c
> MODULE_DESCRIPTION("Multi queue to hardware queue mapping qdisc");

**Comment by DaanDeMeyer** [comment]
> Another option:
> 
>     * `[MultiQueueing]`
> 
>     * `[PriorityMultiQueueing]`
> 
>     * `[BandMultiQueueing]`

Implemented it like this now and added support for `mq` as well.

**Comment by yuwata** [comment]
How about the following?
- `[ClassfulMultiQueueing]` -> mq, the naming 'classful' comes from the first commit when it is introduced, https://github.com/torvalds/linux/commit/6ec1c69a8f6492fd25722f4762721921da074c12
- `[PriorityMultiQueueing]` -> mqprio
- `[MultiQueueing]` or `[BandMultiQueueing]`-> multiq

File: test/test-network/systemd-networkd-tests.py [hunk 1]
```diff
@@ -4648,6 +4648,16 @@ def test_qdisc_ingress(self):
...
 
+    @expectedFailureIfModuleIsNotAvailable('sch_multiq')
+    def test_qdisc_multiq(self):
+        copy_network_unit('25-tun.netdev', '25-tap.netdev', '25-qdisc-multiq.network')
+        start_networkd()
+        self.wait_online('testtun99:degraded', 'testtap99:degraded')
+
+        output = check_output('tc qdisc show dev testtun99')
+        print(output)
+        self.assertRegex(output, 'qdisc multiq 2: root')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: For newer test code, please use `self.assertIn()`

File: test/test-network/systemd-networkd-tests.py [hunk 2]
```diff
@@ -4648,6 +4648,15 @@ def test_qdisc_ingress(self):
...
         self.assertRegex(output, 'qdisc ingress')
 
+    def test_qdisc_mq(self):
+        copy_network_unit('25-tun.netdev', '25-tap.netdev', '25-qdisc-mq.network')
+        start_networkd()
+        self.wait_online('testtun99:degraded', 'testtap99:degraded')
+
+        output = check_output('tc qdisc show dev testtun99')
+        print(output)
+        self.assertRegex(output, 'qdisc mq 2: root')
```
Comment (yuwata) [inline test/test-network/systemd-networkd-tests.py]: Also here.

**Comment by DaanDeMeyer** [comment]
Renamed and addressed comments, setting the green label

**Pull request merged**